Sources/TuistKit/Services/FetchService.swift

```diff
@@ -63,12 +63,6 @@ final class FetchService {
     }
 
     private func fetchDependencies(path: AbsolutePath, update: Bool, with plugins: TuistGraph.Plugins) throws {
-        guard FileHandler.shared.exists(
```

Comment:
Fetch is also for plugins, not just dependencies.
I think we should just make the command fail if we can't find the Project.swift file

Reply:
fetchPlugins relies on `configLoader` and `pluginService` to load plugins and will fail if there are issues

Reply:
Apart from plugins error handling, we could have `Workspace.swift` instead of `Project.swift. I think all that is pretty well handled by the dedicated services with their readable errors

Reply:
Oh maybe it's changed recently, anyway I would make it fail if not manifest is found (either project or workspace), there should already be a method for that if I recall correctly

Reply:
That'll happen inside a `DependenciesModelLoader` `loadDependencies` method.`manifestLoader` will try to load the manifest and will fail the same way `tuist generate` does through `Generator` and `ManifestGraphLoader`

Reply:
Yes but it would make tuist fetch fail on projects without dependencies, and that's potentially a breaking change (e.g. someone might have some integration running tuist fetch in all their projects, regardless from the presence of dependencies)

Reply:
Makes sense, I'll change that, thank you

Reply:
I'm not sure if it's a good idea to use the whole manifestGraph loader, I could just use these lines inside a method

```swift
let manifests = manifestLoader.manifests(at: path)
guard manifests.contains(.workspace) || manifests.contains(.project) else {
    throw ManifestLoaderError.manifestNotFound(path)
}
```

But then `ManifestLoaderError` is used outside of its scope

Reply:
I agree we should use ManifestLoading and not ManifestGrahpLoading.
What about adding a method to ManifestLoading?
Something like:
```
    public func validateHasProjectOrWorkspaceManifest(at: path) throws {
        let manifests = manifestLoader.manifests(at: path)
        guard manifests.contains(.workspace) || manifests.contains(.project) else {
            throw ManifestLoaderError.manifestNotFound(path)
        }
    }
```

Reply:
Fair enough. Added the method and tests